pkg/sdk/validation/action/io_validator.go

```diff
@@ -366,3 +363,39 @@ func (c *InputOutputValidator) ifaceHasNoInput(iface *gqlpublicapi.InterfaceRevi
 	}
 	return false
 }
+
+func (c *InputOutputValidator) implHasNoInputParams(impl gqlpublicapi.ImplementationRevision) bool {
```

Comment:
Suggestion:
I would rename the methods and also invert them for better readability:

```
implHasNoInputParams -> hasInputParams
implHasNoInputInstances -> hasInputInstances
```

But this is something completely optional 🙂

Reply:
`impl` prefix needs to be there as it's the same pkg and it will conflict with `iface`. About changing it from `hasNoInput` to `hasInput` would require `if !implHasInputParams()` which IMO with this `!` is another thing that you need to translate in you mind. If this is not a problem I will leave it as it is 👍